### ISSUE TITLE
Suggestion: Add husky to catch test failures and lint errors locally

### ISSUE THREAD

**Issue by lukeecart** [Codecademy/docs#264]
[Husky](https://typicode.github.io/husky/#/?id=automatic-recommended) is a package that can run git hooks before committing to a repository. This package could also be used to lint commit messages and run other bits of code as well.

Recently there have been a few issues with passing the lint and formating tests. The errors only show after the commit has been pushed and ran on Circle CI. Maybe it would be helpful to run tests, formatter, and lint just as contributors commit?

This can be version controlled because when a contributor runs `yarn install` this husky is initiated. Husky is used on a number of other open-source projects and I think Codecademy docs could benefit from also including it.

**TLDR; Husky will catch errors locally before being run on Circle CI.**

@sonnynomnom What do you think?
